File: 11278-one-handed-typist/Main.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        String qwertyLayout = "`1234567890-=qwertyuiop[]\\asdfghjkl;'zxcvbnm,./ ";
        String shiftQwertyLayout = "~!@#$%^&*()_+QWERTYUIOP{}|ASDFGHJKL:\"ZXCVBNM<>?";
        String qwerty = qwertyLayout + shiftQwertyLayout;

        String dvorakLayout = "`123qjlmfp/[]456.orsuyb;=\\789aehtdck-0zx,inwvg' ";
        String shiftDvorakLayout = "~!@#QJLMFP?{}$%^>ORSUYB:+|&*(AEHTDCK_)ZX<INWVG\"";
        String dvorak = dvorakLayout + shiftDvorakLayout;

        Map<Character, Character> qwertyToDvorak = new HashMap<>();
        for (int i = 0; i < qwerty.length(); i++) {
            qwertyToDvorak.put(qwerty.charAt(i), dvorak.charAt(i));
        }

        Scanner scanner = new Scanner(System.in);

        while (scanner.hasNextLine()) {
            String line = scanner.nextLine();
            StringBuilder out = new StringBuilder();

            for (char ch : line.toCharArray()) {
                out.append(qwertyToDvorak.getOrDefault(ch, ch));
            }

            System.out.println(out);
        }
        scanner.close();
    }
}

File: 11278-one-handed-typist/main.py
import sys

qwerty_layout = "`1234567890-=qwertyuiop[]\\asdfghjkl;'zxcvbnm,./ "
shift_qwerty_layout = "~!@#$%^&*()_+QWERTYUIOP{}|ASDFGHJKL:\"ZXCVBNM<>?"
qwerty = list(qwerty_layout + shift_qwerty_layout)

dvorak_layout = "`123qjlmfp/[]456.orsuyb;=\\789aehtdck-0zx,inwvg' "
shift_dvorak_layout = "~!@#QJLMFP?{}$%^>ORSUYB:+|&*(AEHTDCK_)ZX<INWVG\""
dvorak = list(dvorak_layout + shift_dvorak_layout)

qwerty_to_dvorak = {q: d for q, d in zip(qwerty, dvorak)}

lines = sys.stdin.read().splitlines()

for line in lines:
    out = [qwerty_to_dvorak[ch] if ch in qwerty_to_dvorak else ch for ch in line]
    print(''.join(out))